File: src/main/java/no/nav/familie/ks/sak/app/rest/Behandling/grunnlag/søknad/RestBarn.java
package no.nav.familie.ks.sak.app.rest.Behandling.grunnlag.søknad;

import no.nav.familie.ks.sak.app.behandling.domene.kodeverk.BarnehageplassStatus;
import no.nav.familie.ks.sak.app.behandling.domene.typer.AktørId;

import java.time.LocalDate;

public class RestBarn {
    public AktørId aktørId;
    public BarnehageplassStatus barnehageStatus;
    public Double barnehageAntallTimer;
    public LocalDate barnehageDato;
    public String barnehageKommune;

    public RestBarn(AktørId aktørId, BarnehageplassStatus barnehageStatus, Double barnehageAntallTimer, LocalDate barnehageDato, String barnehageKommune) {
        this.aktørId = aktørId;
        this.barnehageStatus = barnehageStatus;
        this.barnehageAntallTimer = barnehageAntallTimer;
        this.barnehageDato = barnehageDato;
        this.barnehageKommune = barnehageKommune;
    }
}

File: src/main/java/no/nav/familie/ks/sak/app/rest/behandling/grunnlag/søknad/RestBarn.kt
package no.nav.familie.ks.sak.app.rest.behandling.grunnlag.søknad

import no.nav.familie.ks.sak.app.behandling.domene.grunnlag.barnehagebarn.Barn
import no.nav.familie.ks.sak.app.behandling.domene.kodeverk.BarnehageplassStatus
import java.time.LocalDate

data class RestBarn(
    val fødselsnummer: String,
    val barnehageStatus: BarnehageplassStatus,
    val barnehageAntallTimer: Double?,
    val barnehageDato: LocalDate?,
    val barnehageKommune: String?)

fun Barn.toRestBarn() = RestBarn(
    fødselsnummer = this.fødselsnummer,
    barnehageStatus = this.barnehageStatus,
    barnehageAntallTimer = this.barnehageAntallTimer,
    barnehageDato = this.barnehageDato,
    barnehageKommune = this.barnehageKommune
)
